### ISSUE TITLE
shell connections are bounced ~30 seconds into debian boot cycle subsequent to first boot/provision

### ISSUE THREAD

**Issue by mhio** [lima-vm/lima#1941]
### Description

After the first boot/provision process has completed on a debian12 vm. Creating a new lima shell early in any subsequent boot process, the shell is bounced after about the same time as the initial `limactl start` creation command takes. 

limactl version 0.18.0

Watching the systemd journal before the shell session is bounced, it always exits here
```
$ lima sudo journalctl -f
<snip>
Oct 24 00:51:23 debvm cloud-init[1583]: Cloud-init v. 22.4.2 running 'modules:final' at Tue, 24 Oct 2023 00:51:23 +0000. Up 36.11 seconds.
```
Looking through the journal around then, I believe the `boot.sh` script output isn't flushed before termination so it's a bit later the session is terminated:
```
Oct 24 00:51:23 debvm cloud-init[1583]: + command -v loginctl
Oct 24 00:51:23 debvm cloud-init[1583]: + loginctl terminate-user mhio
Oct 24 00:51:23 debvm systemd[1]: Stopping session-2.scope - Session 2 of User mhio...
Oct 24 00:51:23 debvm sshd[1271]: pam_unix(sshd:session): session closed for user mhio
Oct 24 00:51:23 debvm sudo[1286]: pam_unix(sudo:session): session closed for user root
```
Which led me to 
https://github.com/lima-vm/lima/blob/b5aa0dd2f504a189382fb0ea6710a37151407abc/pkg/cidata/cidata.TEMPLATE.d/boot/07-etc-environment.sh#L17

I believe the purpose here is to inject environment before the lima provisioning starts. I assume some config changes might also need to be reflected here after first provision/boot? 

Could the [`login-ctl terminate-user`](https://github.com/lima-vm/lima/blob/b5aa0dd2f504a189382fb0ea6710a37151407abc/pkg/cidata/cidata.TEMPLATE.d/boot/07-etc-environment.sh#L7-L10) be guarded on the content of `/etc/environment` changing? Something like
```
if [ -e /etc/environment ]; then
  etc_environment_sum_pre="$(md5sum /etc/environment)"
  sed -i '/#LIMA-START/,/#LIMA-END/d' /etc/environment
fi
cat "${LIMA_CIDATA_MNT}/etc_environment" >>/etc/environment
etc_environment_sum_post="$(md5sum /etc/environment)"

if command -v loginctl >/dev/null 2>&1 && [ "$etc_environment_sum_pre" != "$etc_environment_sum_post" ]; then
  loginctl terminate-user "${LIMA_CIDATA_USER}" || true
fi
```

lima.yaml
```
images:
  - location: 'https://cloud.debian.org/images/cloud/bookworm/20231013-1532/debian-12-genericcloud-amd64-20231013-1532.qcow2'
    arch: 'x86_64'
    digest: 'sha512:b2ddc01e8d13dabbcfde6661541aae92219be2d442653950f0e44613ddebaeb80dc7a83e0202c5509c5e72f4bd1f4edee4c83f35191f2562b3f31e20e9e87ec2'
  - location: 'https://cloud.debian.org/images/cloud/bookworm/latest/debian-12-genericcloud-amd64.qcow2'
    arch: 'x86_64'
vmType: 'vz'
mountType: 'virtiofs'
mounts:
  - location: '/tmp/lima'
    writable: true
networks:
  - socket: '/private/var/run/socket_vmnet'
containerd:
  system: false
  user: false
hostResolver:
  enabled: true
```

**Comment by mhio** [comment]
Also, the lima guest agent socket removal past first provisioning would be removing the already started `lima-guestagent.service` socket.
https://github.com/lima-vm/lima/blob/b5aa0dd2f504a189382fb0ea6710a37151407abc/pkg/cidata/cidata.TEMPLATE.d/boot/07-etc-environment.sh#L20-L21

Additionally the path to the file is `/run/lima-guestagent.sock` on this debian vm, so maybe isn't doing anything?
```
$ ls -l /run/lima-guestagent.sock 
srwxrwxrwx 1 root root 0 Oct 24 00:51 /run/lima-guestagent.sock
```

**Comment by mhio** [comment]
The disconnect seems to be highlighted due to my use case of not using `limactl start` and waiting, but running the vm's command `/usr/local/bin/limactl hostagent --pidfile /Users/kimi/.lima/default/ha.pid --socket /Users/kimi/.lima/default/ha.sock default` directly via a `launchd` agent. 

The vm also has a slightly longer systemd startup time, waiting for all the containers to be running before lima's cloud init steps run.

**Comment by jandubois** [comment]
> @jandubois Could you take a look?

Sure. In general I would say that this is expected if you connect to an instance before all the "requirements" have been completed that are part of the regular `limactl start` process.

However, I think we should implement the change proposed by @mhio that the user session is not bounced when `/etc/environment` hasn't changed.

@mhio Let me know if you want to create a PR, or if I should do it based on your suggestion above?

**Comment by mhio** [comment]
It looks like the cidata iso is all [built in process](https://github.com/lima-vm/lima/blob/bfd74711f565fe4cbd9cf923b96e6a11e4c785ca/pkg/cidata/template.go#L17), I don't have a build environment to test it right now but commits are there if needed.

**Comment by mhio** [comment]
PR created

The socket removal can left behind if needed,  I kept it in a second commit as it was just clean up and doesn't affect this issue. 
That socket file on my system is `/run/lima-guestagent.sock.  In normal operation, after first boot provisioning, removal of the corrected socket path would remove the already running `lima-guestagent` services socket. So I don't think it's fixing anything in provisioning at the moment, and would only degrade the normal boot up situation if the `rm` was updated to remove the correct path.
